packages/front-end/components/Experiment/ResultsDownloadButton.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  ExperimentReportResultDimension,
+  ExperimentReportVariation,
+} from "back-end/types/report";
+import React, { useMemo } from "react";
+import { FaFileExport } from "react-icons/fa";
+import { useDefinitions } from "../../services/DefinitionsContext";
+import { getRisk } from "../../services/experiments";
+import { Parser } from "json2csv";
+import { MetricInterface, MetricStats } from "back-end/types/metric";
+
+type CsvRow = {
+  date?: string;
+  dimension?: string;
+  metricName?: string;
+  variantName?: string;
+  riskOfChoosing?: number;
+  users?: number;
+  count?: number;
+  conversionRate?: number;
+  chanceToBeatControl?: number | null;
+  percentChange?: number | null;
+};
+
+type Variation = {
+  buckets?: { x: number; y: number }[];
+  ci?: [number, number];
+  cr: number;
+  risk?: [number, number];
+  stats?: MetricStats;
+  users: number;
+  value: number;
+};
+
+type Row = {
+  label?: string;
+  metric?: MetricInterface;
+  rowClass?: string;
+  variations?: Variation[];
+};
+
+export default function ResultsDownloadButton({
+  results,
+  metrics,
+  variations,
+  trackingKey,
+  dimension,
+}: {
+  results?: ExperimentReportResultDimension[];
+  metrics: string[];
+  variations: ExperimentReportVariation[];
+  trackingKey: string;
+  dimension: string;
+}) {
+  const { getMetricById, getDimensionById, ready } = useDefinitions();
+
+  const json2csvParser = new Parser();
+
+  const csvRows: CsvRow[] = [];
+
+  const dimensionName = getDimensionById(dimension);
```

Comment:
This variable type is `DimensionInterface | null`, but based on the variable name I would expect it to be a string.

Also, confusingly, there are 3 types of dimensions:
- Built-in dimension (id = `pre:date` or `pre:activation`)
- Experiment dimensions (id starts with `exp:`)
- User dimensions

You can see the code on the back-end to parse these: https://github.com/growthbook/growthbook/blob/main/packages/back-end/src/services/experiments.ts#L546

`getDimensionById` will only work when it's a User Dimension.  So the actual code here to get the name will have to be a little more complicated.

```ts
const dimensionName =
  getDimensionById(dimension)?.name ||
  dimension?.split(":")?.[1] ||
  dimension ||
  null;
```